spec/controllers/settings/ldap_test_controller_spec.rb

```diff
@@ -0,0 +1,136 @@
+require "rails_helper"
+require "json"
+
+RSpec.describe Settings::LdapTestController, type: :controller do
+    let(:user) { create(:user) }
+  
+    before do
+      setup_done
+      sign_in user
+    end
+
+    describe "POST #create" do
+        let(:host) { "ldaptest.com" }
+        let(:port) { 636 }
+        let(:start_tls) { "false" }
+        let(:cert) { "MIIC0zCCAlmgAwIBAgIUCfQ+m0pgZ/BjYAJvxrn/bdGNZokwCgYIKoZIzj0EAwMw
+            gZYxCzAJBgNVBAYTAlVTMRUwEwYDVQQKEwxBMUEgQ2FyIFdhc2gxJDAiBgNVBAsT
+            G0luZm9ybWF0aW9uIFRlY2hub2xvZ3kgRGVwLjEUMBIGA1UEBxMLQWxidXF1ZXJx
+            dWUxEzARBgNVBAgTCk5ldyBNZXhpY28xHzAdBgNVBAMTFmRvY2tlci1saWdodC1i
+            YXNlaW1hZ2UwHhcNMTUxMjIzMTM1MzAwWhcNMjAxMjIxMTM1MzAwWjCBljELMAkG
+            A1UEBhMCVVMxFTATBgNVBAoTDEExQSBDYXIgV2FzaDEkMCIGA1UECxMbSW5mb3Jt
+            YXRpb24gVGVjaG5vbG9neSBEZXAuMRQwEgYDVQQHEwtBbGJ1cXVlcnF1ZTETMBEG
+            A1UECBMKTmV3IE1leGljbzEfMB0GA1UEAxMWZG9ja2VyLWxpZ2h0LWJhc2VpbWFn
+            ZTB2MBAGByqGSM49AgEGBSuBBAAiA2IABMZf/12pupAgl8Sm+j8GmjNeNbSFAZWW
+            oTmIvf2Mu4LWPHy4bTldkQgHUbBpT3xWz8f0lB/ru7596CHsGoL2A28hxuclq5hb
+            Ux1yrIt3bJIY3TuiX25HGTe6kGCJPB1aLaNmMGQwDgYDVR0PAQH/BAQDAgEGMBIG
+            A1UdEwEB/wQIMAYBAf8CAQIwHQYDVR0OBBYEFE+l6XolXDAYnGLTl4W6ULKHrm74
+            MB8GA1UdIwQYMBaAFE+l6XolXDAYnGLTl4W6ULKHrm74MAoGCCqGSM49BAMDA2gA
+            MGUCMQCXLZj8okyxW6UTL7hribUUbu63PbjuwIXnwi420DdNsvA9A7fcQEXScWFL
+            XAGC8rkCMGcqwXZPSRfwuI9r+R11gTrP92hnaVxs9sjRikctpkQpOyNlIXFPopFK
+            8FdfWPypvA==" }
+        let(:anon_bind) { "false" }
+        let(:dN) { "cn=admin,dc=ldaptest,dc=com" }
+        let(:pass) { "admin" }
+        let(:baseDN) { "dc=ldaptest,dc=com" }
+        let(:filter) { "(objectClass=person)" }
+
+        
+        it "has the expected response status" do
+            post :create, host: host, port: port, start_tls: start_tls, cert: cert, 
+            anon_bind: anon_bind, dN: dN, pass: pass, baseDN: baseDN, filter: filter
+            expect(response).to have_http_status(:ok)
+        end
+
+        it "has the expected result status" do
+            post :create, host: host, port: port, start_tls: start_tls, cert: cert, 
+            anon_bind: anon_bind, dN: dN, pass: pass, baseDN: baseDN, filter: filter
+            parse_json = JSON(response.body)
+            expect(parse_json["result"]["status"]).to eq("pass")
```

Comment:
As stated before, a boolean 'passed' would be sensible for JSON return.
Also it may be reasonable to use http status code to indicate a pass or fail.